Sources/Extension.swift

```diff
@@ -26,6 +26,10 @@ open class Extension {
 
   /// Registers a template filter with the given name
   public func registerFilter(_ name: String, filter: @escaping (Any?, [Any?]) throws -> Any?) {
```

Comment:
Should we deprecate this function (and mark it with `@available(*, deprecated`)?

Reply:
I'm not saying remove it, just mark it as `deprecated`.

Note: having 2 methods that only differ in the parameters of a callback block is quite annoying for Xcode autocomplete, which doesn't handle this very well (at all).

Reply:
Yeah Xcode shouldn't limit our API, but Swift should. And its type inference can sometimes go bonkers on cases like these 🤷‍♂️

Reply:
Maybe, but I didn't came across anything weird when working on that, so I guess it should be file to leave it. Also overload with `context` argument is something most of filters do not need.

Reply:
Ok on this point.